### ISSUE TITLE
F0.5 is almost zero after training the model 

### ISSUE THREAD

**Issue by alsabahi** [grammarly/gector#27]
Thanks for the interesting work. 
**I have trained the model, stage 1, on the first part of the PIE data (a1). After shuffling and splitting the data, I  used the following command:** 

python train.py --train_set "./data/a1/train.txt" --dev_set "./data/a1/dev.txt" --model_dir "./modelxlnet/" --vocab_path=data/output_vocabulary --skip_correct=1 --tn_prob=0 --special_tokens_fix=0 --cold_steps_count=2 --transformer_model=xlnet  --updates_per_epoch=10000 --batch_size=64 --accumulation_size=4 --cold_steps_count=2. 

I have also used the vocab_path from your project's ./data/output_vocabulary.
But after training for more than 5 epochs, I am still getting almost the same input file. 
Here is the command I used for inference: 

_python predict.py --model_path "./modelxlnet/model_state_epoch_2.th" --subset valid  --min_error_probability 0 --additional_confidence 0 --batch_size=64 --iteration_count=5 --transformer_model=xlnet --special_tokens_fix=0_

And here is the ERRANT evaluation result :
![image](https://user-images.githubusercontent.com/33044055/87893617-17dcef80-ca49-11ea-8662-ea8116600d35.png)

Am I missing something. Thanks for the time.

**Comment by komelianchuk** [comment]
Hi, @alsabahi 
Just for clarification:
1. Have you preprocessed data with preprocessing script?
2. Did you observe loss going down during the first 5 epoch? Can you provide loss values after each epoch?
3. What is the `subset` parameter? ( `predict.py` script does not have such one)

**Comment by alsabahi** [comment]
> Hi, @alsabahi
> Just for clarification:
> 
> 1. Have you preprocessed data with preprocessing script?
> 2. Did you observe loss going down during the first 5 epoch? Can you provide loss values after each epoch?
> 3. What is the `subset` parameter? ( `predict.py` script does not have such one)

@komelianchuk  Thanks for the reply. 
1. I have used the preprocessing script provided in your project with the following command:
_python utils/preprocess_data.py -s "./data/a1/a1_train_incorr_sentences.txt"   -t "./data/a1/a1_train_corr_sentences.txt"  -o "./data/a1/a1_train_all_sentences.txt"_
after that I shuffled the file and split into train/dev with 98/2.
2. The loss was decreasing and accuracy was increasing. here is a snapshot of the training logs, sorry,it looks ugly:
![image](https://user-images.githubusercontent.com/33044055/88072331-9470ea80-cb7d-11ea-9e63-20d0767ec188.png)

3. I am sorry about the `subset` parameter, you can ignore it; it's added to get the paths for BEA dev set and test set. Here is the command for inference again:
_python predict.py --model_path "./modelxlnet/model_state_epoch_5.th" --min_error_probability 0 --additional_confidence 0 --batch_size=64 --iteration_count=5 --transformer_model=xlnet --special_tokens_fix=0 --input_file "./data/bea19/ABCN.bea19.dev.orig.txt"  --output_file "./data/ABCN.bea19.dev.cor"_

**Comment by komelianchuk** [comment]
Your loss values are pretty high, so it seems like you have a bug somewhere and it's hard for me to detect the exact reason.
Btw, you can check `metrics_epoch_` files and get loss values from there.
After the first epoch you should get approximately the following.
```
  "best_epoch": 0,
  "peak_cpu_memory_MB": 4159.368,
  "peak_gpu_0_memory_MB": 1656,
  "training_duration": "5:00:14.136983",
  "training_start_epoch": 0,
  "training_epochs": 0,
  "epoch": 0,
  "training_accuracy": 0.9017781938670396,
  "training_loss": 1.041968622905761,
  "training_cpu_memory_MB": 4159.368,
  "training_gpu_0_memory_MB": 1656,
  "validation_accuracy": 0.9040708947806015,
  "validation_loss": 1.1069392061680556,
  "best_validation_accuracy": 0.9040708947806015,
  "best_validation_loss": 1.1069392061680556
```

Please check that you use correct versions of transformer/pytorch/allennlp libraries (as it set in requirements.txt).
Have you modified the code somehow?

**Comment by alsabahi** [comment]
@komelianchuk  I have installed the libraries using the pip command with the requirement file in your project. I haven't modified the code. I am using a fresh copy of your last repository.

**Comment by alsabahi** [comment]
Here is the loss value from metrics_epoch_0 file:
```
"best_epoch": 0,
  "peak_cpu_memory_MB": 4072.136,
  "peak_gpu_7_memory_MB": 5125,
  "training_duration": "5:11:21.513482",
  "training_start_epoch": 0,
  "training_epochs": 0,
  "epoch": 0,
  "training_accuracy": 0.9017559199078139,
  "training_loss": 1.0426828745149077,
  "training_cpu_memory_MB": 4072.136,
  "training_gpu_7_memory_MB": 5125,
  "validation_accuracy": 0.9035023042829239,
  "validation_loss": 1.122864973077178,
  "best_validation_accuracy": 0.9035023042829239,
  "best_validation_loss": 1.122864973077178
```

**Comment by alsabahi** [comment]
Problem solved. I have reinstalled all the dependencies and start fresh. That solved the problem. 

Thanks.
